app/code/Academy/SmsSubscription/Model/SubscriptionManager.php

```diff
@@ -59,4 +59,9 @@ public function updateSubscription(int $id, int $status): void
         $subscription->setData('updated_at', $date->format('Y-m-d H:i:s'));
         $this->subscriptionResource->save($subscription);
     }
+
+    public function deleteSubscription(Subscription $subscription): void
+    {
+        $this->subscriptionResource->delete($subscription);
```

Comment:
`delete()` method throws an exception in case of some error during the delete process so it would be good to handle it, especially that `deleteSubscription()` is used in a loop and one exception not handled will stop further processing of mass delete.